tests/cmdline/params/options/test_interactive.py

```diff
@@ -95,6 +110,20 @@ def test_callback_prompt_twice(self):
         self.assertIn(expected_3, lines[9])
         self.assertIn(expected_4, lines[12])
 
+    def test_callback_prompt_only_once(self):
+        """
+        scenario: using InteractiveOption with type=float and callback that echos an additional message
+        behaviour: the callback should be called at most once per prompt
+        """
+        cmd = self.simple_command(type=float, callback=validate_positive_number_with_echo)
+        runner = CliRunner()
+        result = runner.invoke(cmd, [], input='string\n-1\n-1\n1\n')
```

Comment:
If you accept my other suggestions, the input should also be updated.
```suggestion
        result = runner.invoke(cmd, [], input='string\n-1\n1\n')
```